api/pkg/s3/datastore/gcp/gcp.go

```diff
@@ -318,3 +318,12 @@ func (ad *GcpAdapter) ListParts(listParts *pb.ListParts, context context.Context
 		return listPartsOutput, NoError
 	}
 }
+
+func (ad *GcpAdapter) ListBackendObjects(context context.Context, prefix string, limit int64, marker string) (*pb.ListObjectResponse, S3Error) {
+	log.Logf("list objects of backend[type=%s] is not provided\n", ad.backend.Type)
+	return nil, InternalError
```

Comment:
better return UNSUPPORTED_OPERATION or something similar, instead of InternalError in these cases.

Reply:
This API is not open to the end user yet.